### PR TITLE
 Member conundrumgirl commented 3 hours ago •  Styled Control for form fields.

### PR DESCRIPTION
**Pull request by conundrumgirl**
Styled Control for form fields.
Will need to add validation and change how submission is done

### CODE REVIEW

File: src/components/ProfileValidation/ProfileFieldsEditor.tsx [hunk 1]
```diff
@@ -17,23 +18,30 @@ export type ProfileFieldsEditorProps = {
...
     <>
-      <Box sx={{ marginBottom: theme.spacing(2) }}/*className="ProfileFieldsEditor bootstrap-4-backport ValidationSteps"*/>
-
-        <FormGroup className="required">
-          <FormLabel>First Name</FormLabel>
-          <FormControl
+      <Box
+        sx={{
+          marginBottom: theme.spacing(2),
+        }} /*className=e="ProfileFieldsEditor bootstrap-4-backport ValidationSe="ProfileFieldsEditor bootstrap-4-backport ValidationSe="ProfileFieldsEditor bootstrap-4-backport ValidationSe="ProfileFieldsEditor bootstrap-4-backport ValidationSe="ProfileFieldsEditor bootstrap-4-backport ValidationS"ProfileFieldsEditor bootstrap-4-backport ValidationSteps"*/
```
Comment (jay-hodgson) [inline src/components/ProfileValidation/ProfileFieldsEditor.tsx]: Is this comment useful?

**Review by jay-hodgson — APPROVED** [review]

File: src/components/ProfileValidation/ProfileFieldsEditor.tsx [hunk 2]
```diff
@@ -1,6 +1,13 @@
-import { Box } from '@mui/material'
+import {
+  Box,
+  InputLabel,
+  TextField,
+  formHelperTextClasses,
+  inputBaseClasses,
+} from '@mui/material'
+import { StyledFormControl } from 'components/StyledComponents'
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/ProfileFieldsEditor.tsx:2]: replacing bootstrap with a new component

File: src/components/ProfileValidation/ProfileFieldsEditor.tsx [hunk 3]
```diff
@@ -17,57 +24,88 @@ export type ProfileFieldsEditorProps = {
...
+          </InputLabel>
+          <TextField
+            fullWidth
+            id="firstName"
             onChange={e => {
               verificationSubmission.firstName = e.target.value
               setFirstName(e.target.value)
             }}
             value={firstName}
+            helperText={requiredError}
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/ProfileFieldsEditor.tsx]: this should be conditional once validation is in place

File: src/style/theme.ts [hunk 4]
```diff
@@ -1,84 +1,95 @@
-import { createTheme, Theme } from "@mui/material";
+import { createTheme, Theme } from '@mui/material'
```
Comment (conundrumgirl) [inline src/style/theme.ts:1]: mostly formatting here

File: src/components/ProfileValidation/Attestation.tsx [hunk 5]
```diff
@@ -1,74 +1,86 @@
-
-import { Box } from '@mui/material'
+import { Box, Button } from '@mui/material'
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/Attestation.tsx:1]: - design changes
 - moved continue button and enabled/disabled logic into component

File: src/components/ProfileValidation/ContinueButton.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import ArrowRightAltIcon from '@mui/icons-material/ArrowRightAlt'
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/ContinueButton.tsx:1]: continue is now a shared component wrapping mui Button

File: src/components/ProfileValidation/ORCiDButton.tsx [hunk 7]
```diff
@@ -4,6 +4,7 @@ import { SynapseClient } from 'synapse-react-client'
 import { PROVIDERS } from 'synapse-react-client/dist/containers/Login'
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/ORCiDButton.tsx:4]: design updates

File: src/components/ProfileValidation/ProfileFieldsEditor.tsx [hunk 8]
```diff
@@ -1,11 +1,30 @@
-import { Box } from '@mui/material'
+import { Box, InputLabel, TextField } from '@mui/material'
+import { StyledFormControl } from 'components/StyledComponents'
 import React, { useState } from 'react'
-import { FormControl, FormGroup, FormLabel } from 'react-bootstrap'
+
 import theme from 'style/theme'
 import { VerificationSubmission } from 'synapse-react-client/dist/utils/synapseTypes'
+import { ContinueButton } from './ContinueButton'
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/ProfileFieldsEditor.tsx:7]: adding validation as well as getting the actual verification submission object update (in the parent component)
currently it does not display 'required' error messages just disables continue button if they are not filled in.
It's a small form and it's obvious what needs to be done. I did leave the placeholders to implement the display of error messages if we needed to

File: src/components/ProfileValidation/TermsAndConditionsWrapped.tsx [hunk 9]
```diff
@@ -1,41 +1,62 @@
...
+const TermsAndConditionsWrapped: React.FC<TermsAndConditionsWrappedProps> = ({
+  onNext,
+  verificationSubmission,
+}) => {
+  const [isFormComplete, setIsFormComplete] = React.useState(false)
+
+  return (
+    <Box
+      sx={{
         '&  ul': { margin: theme.spacing(0, -8) },
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/TermsAndConditionsWrapped.tsx:23]: this is probably a temporary solution until we have a firm design. Using specificity to shoehorn the current component into general design theme

File: src/components/ProfileValidation/TermsAndConditionsWrapped.tsx [hunk 10]
```diff
@@ -1,41 +1,62 @@
...
+          fontSize: '14px',
+        },
+      }}
+    >
+      <TermsAndConditions
+        onFormChange={isFormComplete => {
+          setIsFormComplete(isFormComplete)
+        }}
+      />
+      <ContinueButton
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/TermsAndConditionsWrapped.tsx:54]: continue button moved into component

File: src/components/ProfileValidation/VerifyIdentify.tsx [hunk 11]
```diff
@@ -1,54 +1,48 @@
...
-              <button onClick={() => setShowORCiDDialog(true)}>
-                {' '}
-                <img src={EditIcon} alt="edit icon" />
-              </button>
-            </>
-          )}
-          {!verificationSubmission.orcid && <ORCiDButton />}
-        </div>
-      </div>
+      <Box sx={{ marginBottom: '48px' }}>
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/VerifyIdentify.tsx:24]: design related changes and moving 'continue' into the component

File: src/components/StyledComponents.ts [hunk 12]
```diff
@@ -29,3 +38,51 @@ export const StyledInnerContainer = styled(Box, {
...
+ usage: 
+        <StyledFormControl fullWidth variant="standard" margin="normal">
+          <InputLabel shrink htmlFor="someinput">
+            labelText
+          </InputLabel>
+          <TextField
+            id="someinput"/>
+        </StyledFormControl>
+*/
+export const StyledFormControl = styled(FormControl, {
```
Comment (conundrumgirl) [inline src/components/StyledComponents.ts:52]: wrapping mui TextField and InputLabel into StyledFormControl creates the bootstrap look

File: src/components/ProfileValidation/ProfileValidation.tsx [hunk 13]
```diff
@@ -232,7 +232,8 @@ export enum ValidationWizardStep {
 
 function BodyControlFactory(args: {
   step: ValidationWizardStep
-  onFormChange: (a: boolean) => void
+
```
Comment (conundrumgirl) [inline src/components/ProfileValidation/ProfileValidation.tsx:235]: Tacking a couple of issues here:
 - with the current architecture the enabled/disabled state of the continue button did not persist during 'back' navigation
 - verification submission did not get updated.
 Now 'continue/submit' button is it's own dumb component which gets included into the individual subcontrols

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**